scylla/src/transport/connection_pool.rs

```diff
@@ -243,6 +264,83 @@ impl NodeConnectionPool {
     }
 }
 
+struct Keepaliver {
+    connections: Arc<ArcSwap<MaybePoolConnections>>,
+    node_address: IpAddr, // This address is only used to enrich the log messages
+    keepalive_interval: Duration,
+}
+
+impl Keepaliver {
+    pub fn load_connections(&self) -> Vec<Arc<Connection>> {
+        use MaybePoolConnections::*;
+        use PoolConnections::*;
+
+        let pool = self.connections.load_full();
+        match &*pool {
+            Ready(NotSharded(conns)) => conns.clone(),
+            Ready(Sharded { connections, .. }) => connections.iter().flatten().cloned().collect(),
+            Initializing => vec![],
+            Broken => {
+                debug!("Cannot send connection keepalives for node {} as there are no alive connections in the pool", self.node_address);
+                vec![]
+            }
+        }
+    }
+
+    async fn work(self) {
+        let mut interval = tokio::time::interval(self.keepalive_interval);
+        interval.tick().await; // Use up the first, instant tick.
+
+        // Default behaviour (Burst) is not suitable for sending keepalives.
+        use tokio::time::MissedTickBehavior::Delay;
```

Comment:
nitpick: it looks like you imported the definition of `tokio::time::MissedTickBehavior::Delay` to ultimately just use it once, below. Let's just inline it.